src/core/annotation.js

```diff
@@ -2278,6 +2291,43 @@ class SquareAnnotation extends MarkupAnnotation {
     super(parameters);
 
     this.data.annotationType = AnnotationType.SQUARE;
+
+    if (!this.appearance) {
```

Comment:
*Just to help me understand the differences between Annotation types:*

It seems that a few of existing `.... extends MarkupAnnotation`-cases only does this when `quadPoints` is defined, note [e.g. `HighlightAnnotation`](https://github.com/mozilla/pdf.js/blob/8e74278b65d2abf7ee720aa615af87757b05626e/src/core/annotation.js#L2364-L2370), hence I wonder what's different between the various Annotation types here?

Reply:
Good question! I looked into the specification and found that the existing annotation types that support this, namely highlight/squiggly/underline/strikethrough, are text markup annotations and those are _required_ to have quadpoints; see https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf#page=411&zoom=auto,-215,424. Circle and square annotations on the other hand are regular markup annotations and for those quadpoints are not defined; see https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf#page=411&zoom=auto,-215,424.

In our implementation we don't make this distinction because for the way we work with them this difference is not relevant. The overview at https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf#page=398&zoom=auto,-215,775 shows which annotation types are markup annotations and it also shows that highlight/squiggly/underline/strikethrough annotations belong to the text markup annotation "subcategory". The reason is that they are always applied to text, whereas squares/circles/other markup annotations are not text-related and can be used anywhere in the document.

Reply:
> Hopefully this helps a bit to clear things up :-)

That's indeed *very* helpful, thank you!